R/zchunk_LB171.LC_R_Cropland_Yh_GLU_irr.R

```diff
@@ -33,76 +30,139 @@ module_aglu_LB171.LC_R_Cropland_Yh_GLU_irr_DISABLED <- function(command, ...) {
+    L161.ag_irrProd_Mt_R_C_Y_GLU <- get_data(all_data, "temp-data-inject/L161.ag_irrProd_Mt_R_C_Y_GLU") %>%
+      # The following two lines of code will be removed later, when we're using 'real' data
+      gather(year, value, -GCAM_region_ID, -GCAM_commodity, -GLU) %>%   # reshape
+      mutate(year = as.integer(substr(year, 2, 5)))    # change Xyear to year
+
+    L161.ag_rfdProd_Mt_R_C_Y_GLU <- get_data(all_data, "temp-data-inject/L161.ag_rfdProd_Mt_R_C_Y_GLU") %>%
+      # The following two lines of code will be removed later, when we're using 'real' data
+      gather(year, value, -GCAM_region_ID, -GCAM_commodity, -GLU) %>%   # reshape
+      mutate(year = as.integer(substr(year, 2, 5)))    # change Xyear to year
+
+    L161.ag_irrHA_frac_R_C_GLU <- get_data(all_data, "temp-data-inject/L161.ag_irrHA_frac_R_C_GLU") # No year in this data
+
+    # First, calculate the share of irrigated vs. rainfed cropland in the base year by GCAM region, commodity and GLU.
+    L161.ag_irrHA_frac_R_C_GLU %>%
+      select(GCAM_region_ID, GCAM_commodity, GLU, irrHA_frac) %>%
+      # Get the share of rainfed cropland
+      mutate(rfd_share = 1 - irrHA_frac) ->
+    L171.ag_irrHA_frac_R_C_GLU
+
+    # Second, downscale total harvested cropland to irrigated and rainfed by GCAM region, commodity, year and GLU.
+    # Apply the base year share of irrigated vs. rainfed cropland to all historial periods (due to lack of data indicating otherwise).
+    L122.LC_bm2_R_HarvCropLand_C_Yh_GLU %>%
+      # Match the base year share by GCAM region, commodity, year and GLU.
+      left_join_error_no_match(L171.ag_irrHA_frac_R_C_GLU, by = c("GCAM_region_ID", "GCAM_commodity", "GLU")) %>%
+      # Calculate the irrigated cropland cover in all historical periods.
+      mutate(value = value * irrHA_frac,
+             # Where values are missing, assume none irrigated (all rainfed). This is left as a check; there are no missing currently (5/10/17).
+             value = if_else(is.na(value), 0, value)) %>%
+      select(GCAM_region_ID, GCAM_commodity, GLU, year, value) ->
+      L171.LC_bm2_R_irrHarvCropLand_C_Yh_GLU
+
+    L122.LC_bm2_R_HarvCropLand_C_Yh_GLU %>%
+      # Total cropland cover, will be used for where rainfed share is missing
+      rename(total = value) %>%
+      # Match the base year share by GCAM region, commodity, year and GLU.
+      left_join_error_no_match(L171.ag_irrHA_frac_R_C_GLU, by = c("GCAM_region_ID", "GCAM_commodity", "GLU")) %>%
+      # Calculate the rainfed cropland cover in all historical periods.
+      mutate(value = total * rfd_share,
+             # For the rainfed cropland table, missing values default to total cropland quantities; there are no missing currently (5/10/17).
+             value = if_else(is.na(value), total, value)) %>%
+      select(GCAM_region_ID, GCAM_commodity, GLU, year, value) ->
+      L171.LC_bm2_R_rfdHarvCropLand_C_Yh_GLU
+
+    # Extend irrigated cropland cover data to all years, 1700-2010
```

Comment:
Here, interpolating the data back to 1700 is only to pass the `timeshift` test, but those years are not used for calculation. There is in fact no missing data for the historical years 1971-2010. I've also tried just interpolating between 1971-2010 to double check, and it failed the `timeshift` test on `left_join_error_no_match`.